src/fetchtastic/download/latest_pointer.py

```
@@ -0,0 +1,133 @@
+            )
+            return False
+        if tmp_path.exists() or tmp_path.is_symlink():
+            tmp_path.unlink()
+        os.symlink(
+            target_name,
+            tmp_path,
+            target_is_directory=(parent / target_name).is_dir(),
+        )
+        os.replace(tmp_path, link_path)
+        logger.debug("Updated latest pointer: %s -> %s", link_path, target_name)
+        return True
```

Ruff (0.15.12)

[warning] Consider moving this statement to an `else` block

(TRY300)